saleor/graphql/payment/mutations.py

```diff
@@ -954,6 +954,9 @@ def validate_input(
                 if "cancel" not in available_actions:
                     available_actions.append("cancel")
                 transaction["available_actions"] = available_actions
+        elif "available_actions" in transaction:
```

Comment:
I think it's a bit hard to follow the code because you have to wrap your head around what the previous if is doing and later that elif implicitly makes a statement that if the above one has items this one doesn't even if the key is there.

Maybe we could change it to:
```python
available_actions = transaction.get("available_actions")

if available actions:
...
else:
...
```